src/metabase/api/action.clj

```diff
@@ -36,63 +38,89 @@
    (s/optional-key :parameters) (s/maybe [su/Map])
    (s/optional-key :parameter_mappings) (s/maybe su/Map)})
 
-(defn check-actions-enabled
-  "Check whether Actions are enabled and allowed for the [[metabase.models.database]] with `database-id`, or return a
-  400 status code."
-  [database-id]
-  {:pre [(integer? database-id)]}
-  (let [{db-settings :settings, driver :engine, :as db} (db/select-one Database :id database-id)]
-    ;; make sure the Driver supports Actions.
-    (when-not (driver/database-supports? driver :actions db)
-      (throw (ex-info (i18n/tru "{0} Database {1} does not support actions."
-                                (u/qualified-name driver)
-                                (format "%d %s" (:id db) (pr-str (:name db))))
-                      {:status-code 400, :database-id (:id db)})))
-    (binding [setting/*database-local-values* db-settings]
-      ;; make sure Actions are enabled for this Database
-      (when-not (actions/database-enable-actions)
-        (throw (ex-info (i18n/tru "Actions are not enabled for Database {0}." database-id)
-                        {:status-code 400}))))))
-
 (api/defendpoint GET "/"
   "Returns cards that can be used for QueryActions"
   [model-id]
   {model-id su/IntGreaterThanZero}
-  (action/merged-model-action nil :card_id model-id))
+  (let [model (api/read-check Card model-id)]
+    ;; We don't check the permissions on the actions, we assume they are
+    ;; readable if the model is readable.
+    (action/actions-with-implicit-params [model] :model_id model-id)))
```

Comment:
This is an optimization. We don't check each action, we know that the rules is that they are readable if their model is readable.